### PR TITLE
Add vault legal text and clean up position list

### PR DESCRIPTION
**Pull request by ruixhuang**
Simulator Screenshot - iPhone 16 Pro - 2024-10-31 at 14 58 56
Simulator Screenshot - iPhone 16 Pro - 2024-10-31 at 14 13 26

<br/>

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

**Review by tyleroooo — APPROVED** [review]
not sure who could really review this

File: dydx/dydxPresenters/dydxPresenters/_v4/Vault/Landing/dydxVaultViewBuilder.swift [hunk 1]
```diff
@@ -90,48 +91,61 @@ private class dydxVaultViewBuilderPresenter: HostedViewPresenter<dydxVaultViewMo
...
-            let iconType: PlatformIconViewModel.IconType
-            let tokenUnitPrecision: Int
-            if marketId == "USDC-USD" {
-                iconType = .asset(name: "symbol_USDC", bundle: .dydxView)
-                tokenUnitPrecision = 2
-            } else {
-                iconType = .init(url: URL(string: assetMap[assetId]?.resources?.imageUrl ?? ""), placeholderText: assetId.first?.uppercased())
-                tokenUnitPrecision = marketMap[marketId]?.configs?.displayStepSizeDecimals?.intValue ?? 2
+        let newPositions = vault?.positions?.positions?
+            .sorted { (lhs, rhs) -> Bool in
```
Comment (jaredvu) [inline dydx/dydxPresenters/dydxPresenters/_v4/Vault/Landing/dydxVaultViewBuilder.swift:99]: is lhs/rhs left-hand-side and right-hand-side?

Comment (ruixhuang) [reply dydx/dydxPresenters/dydxPresenters/_v4/Vault/Landing/dydxVaultViewBuilder.swift:99]: Yeah

File: dydx/dydxPresenters/dydxPresenters/_v4/Vault/Landing/dydxVaultViewBuilder.swift [hunk 2]
```diff
@@ -90,48 +91,61 @@ private class dydxVaultViewBuilderPresenter: HostedViewPresenter<dydxVaultViewMo
...
+                    }
+                    return lhs.thirtyDayPnl?.absolute?.doubleValue ?? 0 < rhs.thirtyDayPnl?.absolute?.doubleValue ?? 0
+                }
+                return lhs.marginUsdc?.doubleValue ?? 0 > rhs.marginUsdc?.doubleValue ?? 0
             }
+            .map { (position) -> dydxVaultPositionViewModel? in
+                guard
+                    let marketId = position.marketId,
+                    // special case for fake USDC market to show unused margin
+                    let assetId = marketId == "UNALLOCATEDUSDC-USD" ? "USDC" : marketMap[marketId]?.assetId
```
Comment (jaredvu) [inline dydx/dydxPresenters/dydxPresenters/_v4/Vault/Landing/dydxVaultViewBuilder.swift:112]: Pull strings into enums or top level constants?

**Comment by ruixhuang** [comment]
> not sure who could really review this

Just stamp ;)

**Pull request merged**